Map/MapManager.gd

```diff
@@ -6,14 +6,19 @@ static func create_map(): ## Generates and Populates a map with rooms that have
 	var _grid: Array = [] ## 2d array to return. this will be populated with rooms
 	
 	for index_height: int in range(_map.floors_width.size()): ## loop through elements of the grid and assign a room type
+		var _max_floor_size = _map.floors_width[_map.floors_width.size()/2]
```

Comment:
this can be outside the loop, no need to calculate the max width of the map each time
the max isn't necessarily at the middle of the map either (you could imagine a map that gets big, smaller at the middle, then big again).
For this, i would advise on using the `max` function (which works on array). So probably:
`var _max_floor_size = max(_map.floors_width)`